react/features/lobby/middleware.js

```diff
@@ -118,6 +119,7 @@ function _conferenceFailed({ dispatch, getState }, next, action) {
             hideErrorSupportLink: true,
             titleKey: 'lobby.joinRejectedMessage'
         }));
+        dispatch(cancelKnocking());
```

Comment:
Thanks @gkdre234 for submitting this request. I find this proposal very useful.

Note that dispatching a notification will not wait for the timeout. In order to navigate to the main page after a timeout I have to use `setTimeout(() => dispatch(cancelKnocking()), NOTIFICATION_TIMEOUT * 3);`